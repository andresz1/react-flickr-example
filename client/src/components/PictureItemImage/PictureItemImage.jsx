// @flow

import * as React from 'react';
import styles from './PictureItemImage.css';

type IProps = {
  children?: React.Node,
  src?: string,
  onClick: Function,
};

const PictureItemImage = (props: IProps) => (
  <div
    className={styles.container}
    style={{backgroundImage: (props.src)? `url(${props.src})`: ''}}
    onClick={props.onClick}
  >
    {props.children}
  </div>
);

export default PictureItemImage;
