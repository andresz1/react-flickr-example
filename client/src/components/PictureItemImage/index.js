import PictureItemImage from './PictureItemImage.jsx';

export default PictureItemImage;
