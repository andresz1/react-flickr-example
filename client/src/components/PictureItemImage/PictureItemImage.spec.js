import React from 'react';
import { shallow } from 'enzyme';
import PictureItemImage from './PictureItemImage';

const setup = (setupProps = {}) => {
  const defaultProps = {
    src: 'https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg',
    handleClick: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <PictureItemImage
      src={props.src}
      onClick={props.handleClick}
    />
  );

  return {
    props,
    wrapper
  };
};

describe('<PictureItemImage />', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('puts empty background when image is not provided', () => {
    const { wrapper } = setup({
      src: undefined,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('calls onClick when clicked', () => {
    const { props, wrapper } = setup();

    wrapper.simulate('click');
    expect(props.handleClick).toHaveBeenCalled();
  });
});
