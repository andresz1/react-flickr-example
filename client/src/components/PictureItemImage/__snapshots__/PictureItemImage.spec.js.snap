// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PictureItemImage /> puts empty background when image is not provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PictureItemImage
    onClick={[Function]}
    src={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "container",
      "onClick": [Function],
      "style": Object {
        "backgroundImage": "",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "container",
        "onClick": [Function],
        "style": Object {
          "backgroundImage": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<PictureItemImage /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PictureItemImage
    onClick={[Function]}
    src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "container",
      "onClick": [Function],
      "style": Object {
        "backgroundImage": "url(https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg)",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "container",
        "onClick": [Function],
        "style": Object {
          "backgroundImage": "url(https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
